apps/wappush/js/parsed_message.js

```diff
@@ -96,7 +93,16 @@
      * Returns true if the message has already expired
      */
     isExpired: function pm_isExpired() {
-      return (this.expires && (this.expires < Date.now()));
+      return (this.expires && (this.expires < Date.now())) || false;
+    },
+
+    /**
+     * Gets the message unique identifier
+     *
+     * @returns {String} An identifier for this message
+     */
+    getUniqueId: function pm_getUniqueId() {
+      return this.timestamp;
```

Comment:
I'm still somewhat nervous with using the timestamp as unique id :) but maybe it's good enough